lib/katello/tasks/upgrades/4.8/regenerate_imported_repository_metadata.rake

```diff
@@ -0,0 +1,31 @@
+namespace :katello do
+  namespace :upgrades do
+    namespace '4.8' do
+      desc "Regenerates metadata for library repositories that were imported"
+      task :regenerate_imported_repository_metadata => ["dynflow:client", "environment"] do
+        User.current = User.anonymous_admin #set a user for orchestration
+        versions = Katello::ContentViewVersionImportHistory.all.map(&:content_view_version)
+        repo_ids = versions.map do |ver|
+           rps = if ver.default?
+                   ver.repositories
+                  else
+                    ver.archived_repos
+                  end
+           rps.exportable
+              .map(&:library_instance_or_self)
+              .select(&:using_mirrored_metadata?)
+              .map(&:id)
```

Comment:
Do you need to pluck out the IDs since you'll already have all the repository objects anyway from the select? Then you could just pass the repos into the foreman task bulk metadata generate.  That would avoid the `where` statement.